### PR TITLE
Fixing onSettingsUpdate and shutdown

### PR DESCRIPTION
**Pull request by moizjv**

### CODE REVIEW

**Comment by jlipps** [comment]
why is my commit in here? it's already in master, right?

File: lib/driver.js [hunk 1]
```diff
@@ -83,6 +84,10 @@ class AndroidDriver extends BaseDriver {
     // start UiAutomator
     this.bootstrap = new Bootstrap(DEVICE_PORT, this.opts.websocket);
     await this.bootstrap.start(this.opts.appPackage, this.opts.disableAndroidWatchers);
+    // handling unexpected shutdown
+    this.bootstrap.onUnexpectedShutdown.catch((err) => {
+      this.startUnexpectedShutdown(err);
```
Comment (jlipps) [inline lib/driver.js]: technically this should be `await`ed as part of an async function

File: lib/driver.js [hunk 2]
```diff
@@ -177,11 +182,30 @@ class AndroidDriver extends BaseDriver {
...
   // Set CompressedLayoutHierarchy on the device
   async setCompressedLayoutHierarchy (compress) {
-    await this.bootstrap.sendCommand("compressedLayoutHierarchy", {compressLayout: compress});
+    await this.bootstrap.sendAction("compressedLayoutHierarchy", {compressLayout: compress});
   }
 
-  // TODO: add shutdown flow, unexpected uiautomator exit handling,
-  // call to stopChromedriverProxies, etc...
+  async deleteSession () {
+    if (this.bootstrap) {
```
Comment (jlipps) [inline lib/driver.js]: i think this check should just be around the bootstrap shutdown line.

File: lib/driver.js (hunk 1 above)
Comment (jlipps) [inline lib/driver.js]: don't you need to set `this.bootstrap` to null here as well, since it's shut down at this point?

**Comment by moizjv** [comment]
closing this in favor of #11

**Pull request closed without merging**